src/uu/ls/src/dired.rs

```diff
@@ -42,7 +42,9 @@ use uucore::error::UResult;
 #[derive(Debug, Clone, PartialEq)]
 pub struct BytePosition {
```

Comment:
it is not a big deal because --dired isn't used a lot but we are storing extra data every time even this case happens rarely

Reply:
I know this solution might be a bit silly, but I'm not good at programming.

It seems like 'end_line' is only used to calculate the offset for the next line, so it should be fine to just store 'end_filename' maybe.

I'll try to optimize it. :)